ifrs17/Import/Importers.ipynb

```diff
@@ -1180,13 +1174,53 @@
+                "\n",
+                "\n    if(log.Errors.Any()) return Activity.Finish().Merge(log); ",
+                "\n    var toCommitParameters  = (await workspace.Query<DataNodeParameter>().ToArrayAsync()).Except(committedParameters, ParametersComparer.Instance());",
+                "\n    if (!toCommitParameters.Any()) {",
+                "\n        ApplicationMessage.Log(Warning.VariablesAlreadyImported);   ",
+                "\n        return Activity.Finish().Merge(log);",
+                "\n    }",
+                "\n",
+                "\n    var allArgs = await GetAllArgsAsync(primaryArgs, options.TargetDataSource, ImportFormats.DataNodeParameter);",
+                "\n    var targetDataNodes = toCommitParameters.Select(x => x.DataNode)",
+                "\n                  .Concat(toCommitParameters.Where(x => x is InterDataNodeParameter).Select(x => ((InterDataNodeParameter)x).LinkedDataNode)).ToHashSet();",
+                "\n    var workspaceToCompute = Workspace.CreateNew();",
+                "\n    workspaceToCompute.Initialize(x => x.FromSource(options.TargetDataSource).DisableInitialization<RawVariable>());",
+                "\n",
+                "\n    foreach (var args in allArgs) { ",
+                "\n        await CommitPartitionAsync<PartitionByReportingNodeAndPeriod>(args, workspace, workspaceToCompute);",
+                "\n        var targetPartition = (Guid)(await options.TargetDataSource.Partition.GetKeyForInstanceAsync<PartitionByReportingNodeAndPeriod>(args));",
+                "\n        var defaultPartition = (Guid)(await options.TargetDataSource.Partition.GetKeyForInstanceAsync<PartitionByReportingNodeAndPeriod>(args with {Scenario = null}));",
+                "\n        await options.TargetDataSource.Partition.SetAsync<PartitionByReportingNodeAndPeriod>(null);",
+                "\n        if(ApplicationMessage.HasErrors()) return Activity.Finish().Merge(log);",
+                "\n        ",
+                "\n        // Avoid starting the computation if no best estimate cash flow and actuals has ever been imported ",
+                "\n        if(args.Scenario == null) {",
+                "\n            if(!(await options.TargetDataSource.Query<RawVariable>().Where(x => x.Partition == targetPartition).Take(1).ToArrayAsync()).Any() &&",
+                "\n               !(await options.TargetDataSource.Query<IfrsVariable>().Where(x => x.Partition == targetPartition).Take(1).ToArrayAsync()).Any()) continue;",
+                "\n        }",
+                "\n",
+                "\n        // Only nominals corresponding to the target data nodes are added to the workspace",
+                "\n        var nominals = await options.TargetDataSource.Query<RawVariable>().Where(x => targetDataNodes.Contains(x.DataNode) && (x.Partition == targetPartition || x.Partition == defaultPartition)).ToArrayAsync();",
```

Comment:
don't you need also previous period partition?
This solution seems to me a bit complex... and it is too easy to miss some pieces... also hard to maintain...

Reply:
previous partition added! We can improve this logic with the next PR.